charts/k8s-service/values.yaml

```diff
@@ -168,6 +168,33 @@ canary: {}
 # point in time. For example, setting to 3 will signal Kubernetes (via the Deployment contoller) to maintain 3 pods.
 replicaCount: 1
 
+# deploymentStrategy specifies the strategy used to replace old Pods by new ones. Type can be "RollingUpdate" or
+# "Recreate". "RollingUpdate" is the default value.
+# RollingUpdate: The Deployment updates Pods in a rolling update fashion.
+# Recreate: All existing Pods are killed before new ones are created.
+# 
+# RollingUpdate can be further refined by using the options provided by the rollingUpdate variable.
+# The rollingUpdate variable is a map that is directly injected into the deployment spec and it has the following keys:
+#   - Max Unavailable   (Optional) : Field that specifies the maximum number of Pods that can be unavailable
+#                                    during the update process. The value can be an absolute number
+#                                    (for example, 5) or a percentage of desired Pods (for example, 10%).
+#                                    The value cannot be 0 if rollingUpdate.maxSurge is 0. 
+#                                    This option defaults to 25%.
+#   - Max Surge         (Optional) : Field that specifies the maximum number of Pods that can be created over 
+#                                    the desired number of Pods. The value can be an absolute number (for example, 5)
+#                                    or a percentage of desired Pods (for example, 10%). The value cannot be 0 if 
+#                                    MaxUnavailable is 0. 
+#                                    This option defaults to 25%.
+#
+# EXAMPLE: 
+#
+# deploymentStrategy: RollingUpdate
+# rollingUpdate:
+#   maxSurge: 30%
+#   maxUnavailable: 30% 
+deploymentStrategy: RollingUpdate
+rollingUpdate: {}
```

Comment:
I think it is a slightly better interface to use the `enabled` pattern (and default to `false`). This way, we don't have a diversion problem if Kubernetes decides to update the default deployment strategy in later versions.

Reply:
Hey @MaxKam @yorinasub17, I'm happy to pick this back off if you don't mind as I've run into this limitation (particually need to set `recreate` deployment stragety).

I'm not fully following your suggestion here @yorinasub17 , do you mean adding the enabled flag and inferring deployment strategy based on this object being enabled? (In which case would we just have a simple Recreate object for supporting recreate?)

Reply:
The `enabled` pattern is basically to omit the `strategy` configuration from the resulting object when `false`. E.g., I would change this input value to be:

```
deploymentStrategy:
  enabled: false
  strategy: RollingUpdate
  rollingUpdate: {}
```

And then in the actual pod spec, guard the `strategy` block with `{{- if deploymentStrategy.enabled }}`.